src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.iOS/AppleCertificatePal.cs

```diff
@@ -0,0 +1,572 @@
+            try
+            {
+                unsafe
+                {
+                    fixed (byte* pin = rawData)
+                    {
+                        using (var manager = new PointerMemoryManager<byte>(pin, rawData.Length))
+                        {
+                            PfxAsn.Decode(manager.Memory, AsnEncodingRules.BER);
+                        }
+
+                        return true;
+                    }
+                }
+            }
+            catch (CryptographicException)
+            {
+            }
+
+            return false;
+        }
+
+        private static bool IsPkcs7Signed(ReadOnlySpan<byte> rawData)
+        {
+            try
+            {
+                unsafe
+                {
+                    fixed (byte* pin = rawData)
+                    {
+                        using (var manager = new PointerMemoryManager<byte>(pin, rawData.Length))
+                        {
+                            AsnValueReader reader = new AsnValueReader(rawData, AsnEncodingRules.BER);
+
+                            ContentInfoAsn.Decode(ref reader, manager.Memory, out ContentInfoAsn contentInfo);
+
+                            switch (contentInfo.ContentType)
+                            {
+                                case Oids.Pkcs7Signed:
+                                case Oids.Pkcs7SignedEnveloped:
+                                    return true;
+                            }
+                        }
+                    }
+                }
+            }
+            catch (CryptographicException)
+            {
+            }
+
+            return false;
+        }
+
+        internal static X509ContentType GetDerCertContentType(ReadOnlySpan<byte> rawData)
+        {
+            X509ContentType contentType = Interop.AppleCrypto.X509GetContentType(rawData);
+
+            if (contentType == X509ContentType.Unknown)
+            {
+                if (IsPkcs12(rawData))
+                {
+                    return X509ContentType.Pkcs12;
+                }
+
+                if (IsPkcs7Signed(rawData))
+                {
+                    return X509ContentType.Pkcs7;
+                }
+            }
+
+            return contentType;
+        }
+
+        private static ICertificatePal FromDerBlob(
+            ReadOnlySpan<byte> rawData,
+            SafePasswordHandle password,
+            X509KeyStorageFlags keyStorageFlags)
+        {
+            Debug.Assert(password != null);
+
+            X509ContentType contentType = GetDerCertContentType(rawData);
+
+            if (contentType == X509ContentType.Pkcs7)
+            {
+                throw new CryptographicException(
+                    SR.Cryptography_X509_PKCS7_Unsupported,
+                    new PlatformNotSupportedException(SR.Cryptography_X509_PKCS7_Unsupported));
+            }
+
+            if (contentType == X509ContentType.Pkcs12)
+            {
+                // TODO: keyStorageFlags
```

Comment:
Please ensure there's an issue tracking the work here, and then reference it from the todo.

I know there's something weird on macOS with PersistKeySet + !Exportable (added to the default keychain, IIRC), it maybe needs to happen here, too.

If the keys are/support being ephemeral then they don't need to throw.  (And if we can make macOS keys ephemeral then we can remove the throw there)

Reply:
I guess I should also file issue for Android since it also ignores the flags, right?